src/messages/MOSDOp.h

```diff
@@ -267,7 +273,7 @@ struct ceph_osd_request_head {
   }
 
   virtual void decode_payload() {
-    bufferlist::iterator p = payload.begin();
+    p = payload.begin();
```

Comment:
Before doing this we should probably assert that partialDecodeNeeded and fullDecodeNeeded are set as expected?